### DISCUSSION TITLE
Add beta/alpha/rc tag to version string

### DISCUSSION THREAD

**Discussion by leccelecce** [blakeblackshear/frigate#5748] [General]
Would it be possible to add the beta tag to the `VERSION` string going forward? i.e. "0.12.0" becomes "0.12.0-beta9" or "0.12.0-rc1".

This would mean a) there would be an entry in the commit log to easily see what changes are included in a given beta version b) the System page would tell you what beta version you're on.

Might be a breaking change for API users if they are doing a numeric V1 > V2 check, though easily solved by cutting off at the dash.

Currently I believe there isn't an easy way to do this other than looking at the Docker image tags.

https://github.com/blakeblackshear/frigate/blob/dev/Makefile#L4

```
COMMIT_HASH := $(shell git log -1 --pretty=format:"%h"|tail -1)
VERSION = 0.12.0
IMAGE_REPO ?= ghcr.io/blakeblackshear/frigate
CURRENT_UID := $(shell id -u)
CURRENT_GID := $(shell id -g)

version:
	echo 'VERSION = "$(VERSION)-$(COMMIT_HASH)"' > frigate/version.py
```

**Comment by NickM-27** [comment]
Personally, not sure it's needed. That's what GitHub tags and diffs are for https://github.com/blakeblackshear/frigate/compare/v0.12.0-beta8...v0.12.0-beta9

**Comment by leccelecce** [comment]
That kind of solves a), but not b). I don't think it's unreasonable to want to know, within the application itself, what release it you're running.

**Comment by leccelecce** [comment]
Also, I assume that this information is essentially somewhere already in the form of the git tag? So potentially this could be automated into the release process. I don't know the setup by which Github triggers a build i.e. do you create a tag and then instruct it to build a release off the tag?

**Reply by NickM-27** [reply]
The tag is created when a release is done, and the tag is manually added to the latest build. Multiple builds can also share multiple tags (last beta will be rc1, last RC will also likely be the main release for example) so it's not very simple.
